src/epoch.rs

```diff
@@ -1,19 +1,19 @@
-/// The global epoch
-///
-/// The last bit in this number is unused and is always zero. Every so often the global epoch is
-/// incremented, i.e. we say it "advances". A pinned mutator may advance the global epoch only if
-/// all currently pinned mutators have been pinned in the current epoch.
-///
-/// If an object became garbage in some epoch, then we can be sure that after two advancements no
-/// mutator will hold a reference to it. That is the crux of safe memory reclamation.
+//! The global epoch
+//!
+//! The last bit in this number is unused and is always zero. Every so often the global epoch is
+//! incremented, i.e. we say it "advances". A pinned mutator may advance the global epoch only if
+//! all currently pinned mutators have been pinned in the current epoch.
+//!
+//! If an object became garbage in some epoch, then we can be sure that after two advancements no
+//! mutator will hold a reference to it. That is the crux of safe memory reclamation.
 
 use std::ops::Deref;
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering::{Relaxed, Acquire, Release, SeqCst};
 
 use mutator::LocalEpoch;
 use mutator::Scope;
-use sync::list::{List, IterResult};
+use sync::michael_list::{List, IterResult};
```

Comment:
Honestly, I dislike identifiers like `michael_list`, `TreiberStack`, `MsQueue` and `chase_lev`:

1. Some of these data structures (particularly list and stack) are so simple that associating them with just one of the papers discussing them is sort of odd.
2. Our implementations are not really identical to the ones presented in papers. 
3. `list`, `Stack`, `Queue`, `deque` are easier names to type and remember.
4. Choosing a less specific name doesn't lock you in to a particular implementation. E.g. if you implement a `MsQueue`, but a month later find out how to change the algorithm to be faster, it will probably stat diverging from the original paper by Michael & Scott, but the name will stay `MsQueue`.

Reply:
I renamed them to `list` and `queue`.